content/java/concepts/threading/threading.md

```diff
@@ -0,0 +1,97 @@
+---
+Title: 'Threading'
+Description: 'Threading is the ability for different parts of a program to run simultaneously.'
+Subjects:
+  - 'Code Foundations'
+  - 'Computer Science'
+Tags:
+  - 'Classes'
+  - 'Inheritance'
+  - 'Interface'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+**Threading** is the ability for different parts of a program to run simultaneously. The currently running part of the program can start another process that will run in the background as the code that spawned the background process continues to execute. This allows complex processes to run without holding up the execution of the rest of the program.
+
+Different parts of the program that are operating at the same time are called "threads" and the code that started executing first, starting these other processes, is typically known as the "main thread".
+
+One important thing to be aware of when working with threads is the possibility of concurrency problems, which happen when threads share access to a particular variable. If multiple threads access one variable, no thread can predict what its value will be since each thread operates independently and may be in a different stage in its process at any given time. It is best practice to avoid accessing such variables unless it's confirmed that the threads that can modify them have completed their execution.
```

Comment:
```suggestion
## Race Conditions

Since Java supports running multiple threads at the same time, the risk of causing a concurrency problem, or a race condition, is always there. This happens when a particular [variable](https://www.codecademy.com/resources/docs/java/variables) is being accessed and updated by many threads (which are running independently and may not be finished executing). Such variables should be avoided unless it's confirmed that the threads that can modify them have completed their execution.
```

This particular paragraph deserves its own header, I think.